bootstrap-app/src/cmr/bootstrap/data/bulk_index.clj

```diff
@@ -119,6 +119,39 @@
             gran-count
             provider-id)))
 
+(defn- index-variables-by-provider
+  "Bulk index variables for the given provider."
+  [system db provider]
+  (info "Indexing variables for provider" (:provider-id provider))
+  (let [concept-batches (db/find-concepts-in-batches
+                         db provider
+                         {:concept-type :variable}
+                         (:db-batch-size system))
+        num-variables (index/bulk-index
+                       {:system (helper/get-indexer system)}
+                       concept-batches
+                       {})]
+    (info (format "Indexing of %s variables completed." num-variables))
+    num-variables))
+
+(defn index-variables
+  "Bulk index variables for the given provider-id."
+  [system provider-id]
+  (let [db (helper/get-metadata-db-db system)
```

Comment:
Looks like this will require refactoring `get-provider-by-id` and adding a `get-providers` helper function that does the db lookup from context/system ...